pandapower/converter/jao/from_jao.py

```diff
@@ -0,0 +1,987 @@
+        with open(html_file_path, mode='r', encoding=kwargs.get("encoding", "utf-8")) as f:
+            html_str = f.read()
+    else:
+        html_str = ""
+
+    # --- manipulate data / data corrections
+    if apply_data_correction:
+        html_str = _data_correction(data, html_str)
+
+    # --- parse html_str to line_geo_data
+    line_geo_data = None
+    if html_str:
+        try:
+            line_geo_data = _parse_html_str(html_str)
+        except (json.JSONDecodeError, KeyError, AssertionError) as e:
+            logger.error(f"html data were ignored due to this error:\n{e}")
+
+    # --- create the pandapower net
+    net = create_empty_network(name=os.path.splitext(os.path.basename(excel_file_path))[0],
+                               **{key: val for key, val in kwargs.items() if key == "sn_mva"})
+    _create_buses_from_line_data(net, data)
+    _create_lines(net, data)
+    _create_transformers_and_buses(net, data, **kwargs)
+
+    # --- invent connections between grid groups
+    if extend_data_for_grid_group_connections:
+        _invent_connections_between_grid_groups(net, **kwargs)
+
+    # --- drop islanded grid groups
+    if drop_grid_groups_islands:
+        drop_islanded_grid_groups(net, kwargs.get("min_bus_number", 6))
+
+    # --- add geo data to buses and lines
+    if line_geo_data is not None:
+        _add_bus_geo(net, line_geo_data)
+        set_line_geodata_from_bus_geodata(net)
+
+    return net
+
+# --- secondary functions --------------------------------------------------------------------------
+
+
+def _data_correction(data:dict[str, pd.DataFrame], html_str:str|None) -> str|None:
+    """Corrects input data in particular with regard to obvious weaknesses in the data provided,
+    such as inconsistent spellings and missing necessary information
+
+    Parameters
+    ----------
+    data : dict[str, pd.DataFrame]
+        data provided by the excel file which will be corrected
+    html_str : str | None
+        data provided by the html file which will be corrected
+
+    Returns
+    -------
+    str
+        corrected html_str
+    """
+
+    # --- Line and Tieline data ---------------------------
+    for key in ["Lines", "Tielines"]:
+
+        # --- correct column names
+        cols = data[key].columns.to_frame().reset_index(drop=True)
+        cols.loc[cols[1] == "Voltage_level(kV)", 0] = None
+        cols.loc[cols[1] == "Comment", 0] = None
+        cols.loc[cols[0].str.startswith("Unnamed:").astype(bool), 0] = None
+        cols.loc[cols[1] == "Length_(km)", 0] = "Electrical Parameters"  # might be wrong in
+        # Tielines otherwise
+        data[key].columns = pd.MultiIndex.from_arrays(cols.values.T)
+
+        # --- correct comma separation and cast to floats
+        data[key][("Maximum Current Imax (A)", "Fixed")] = \
+            data[key][("Maximum Current Imax (A)", "Fixed")].replace("\xa0", 999e3).replace(
+                "-", 999e3).replace(" ", 999e3)
```

Comment:
I think overwriting of N/A imax should not be done, or at least be optional and with a warning.
Max. I DLR is often provided for these lines and could serve as a substitute (sth. like a flag `apply_DLRmax_for_missing_max_i`)